src/ts/src/common/utils/tableUtils.tsx

```diff
@@ -20,15 +24,15 @@ function createTableCellRenderer(
   };
 }
 
-function createTreeModalInfo(link: string): ModalInfo {
+export const createTreeModalInfo = memoize(createTreeModalInfo_);
```

Comment:
memoize the created object, so we don't unnecessarily re-render the component. The reason is because React by default does shallow diffing, so if we generate a new object every time, React will re-render!

E.g.,

`createTreeModalInfo_(1) === createTreeModalInfo_(1) // FALSE`
`createTreeModalInfo(1) === createTreeModalInfo(1) // TRUE thanks to memoization`